bots/archive/bot.py

```diff
@@ -0,0 +1,174 @@
+import datetime
+import bz2
+import io
+import logging
+
+import boto3
+import botocore
+
+from django.db import transaction
+from django.conf import settings
+from django.core.paginator import Paginator
+from django.core import serializers
+
+from share.bot import Bot
+from share.models import CeleryProviderTask
+
+logger = logging.getLogger(__name__)
+
+
+def get_bucket(bucket_name):
+    s3 = boto3.resource('s3')
+    try:
+        s3.meta.client.head_bucket(Bucket=bucket_name)
+    except botocore.exceptions.ClientError as e:
+        # If a client error is thrown, then check that it was a 404 error.
+        # If it was a 404 error, then the bucket does not exist.
+        error_code = int(e.response['Error']['Code'])
+        if error_code != 404:
+            raise botocore.exceptions.ClientError(e)
+        s3.create_bucket(Bucket=bucket_name)
+
+    return bucket_name
+
+
+def queryset_to_compressed_json(queryset, model):
+    compressed_output = io.BytesIO()
+    output = io.StringIO(serializers.serialize('json', queryset))
+    compressed_output.write(bz2.compress(str.encode(output.getvalue())))
+
+    return compressed_output
+
+
+def put_s3(bucket, location, data):
+    s3 = boto3.resource('s3')
+    try:
+        current_date = datetime.datetime.utcnow().isoformat()
+        s3.Object(bucket, location + current_date + '.bz2').put(Body=data.getvalue())
```

Comment:
Final request, I promise. change `.bz2` to `.json.bz2`. Django whines when there is no file extension and the archive utility on OSX defaults file names to the archive name minus `.zip/bz/tar.gz`